fsm.h

```diff
@@ -149,23 +149,23 @@ The following example implements this simple state machine.
   void action1() { std::cout << "perform custom action 1\n"; }
   void action2() { std::cout << "perform custom action 2\n"; }
   enum class States {Initial, A, Final};
-  using F = FSM::Fsm<States, States::Initial>;
+  using F = FSM::Fsm<States, States::Initial, char>;
```

Comment:
Instead of using a char as a trigger, I would prefer the example to use a `enum class`. There are some examples of this in the test cases.